compiler/src/dotty/tools/backend/jvm/BCodeSyncAndTry.scala

```diff
@@ -203,6 +208,13 @@ trait BCodeSyncAndTry extends BCodeBodyBuilder {
        */
       val postHandlers = new asm.Label
 
+      // stack stash
+      val needStackStash = !stack.isEmpty && !caseHandlers.isEmpty
+      val acquiredStack = if needStackStash then stack.acquireFullStack() else null
+      val stashLocals =
+        if acquiredStack == null then null
+        else acquiredStack.uncheckedNN.filter(_ != UNIT).map(btpe => locals.makeTempLocal(btpe))
```

Comment:
We track `UNIT`s because it helps being generic. The `genLoad`s that actually generate a `UNIT` can generically `push(UNIT)` and then `pop()`. If we did not track `UNIT`s, we would have to also send the btype to `pop()` to avoid removing anything when it's `UNIT`.

The previous code would also handle that generically with `btype.size`, which returns 0 for `UNIT`. Now it's going to count `UNIT`s as 0's in `heightDiffWrt`.

In `stashLocals` we have to explicitly filter out `UNIT`s because we cannot create `void`-typed local variables.

Reply:
Got it, thanks 👍